jsons/serializers.py

```diff
@@ -158,28 +159,53 @@ def default_object_serializer(obj: object,
     private attributes (i.e. attributes that start with an underscore).
     :param strip_properties: if ``True`` the resulting dict will not contain
     values from @properties.
+    :param strip_class_variables: if ``True`` the resulting dict will not
+    contain values class variables.
     :param kwargs: any keyword arguments that are to be passed to the
     serializer functions.
     :return: a Python dict holding the values of ``obj``.
     """
-    obj_dict = obj.__dict__ if strip_properties and hasattr(obj, '__dict__') \
-        else _get_dict_from_obj(obj, strip_privates, **kwargs)
-    return default_dict_serializer(obj_dict, key_transformer=key_transformer,
+    obj_dict = _get_dict_from_obj(obj, strip_privates, strip_properties,
+                                  strip_class_variables, **kwargs)
+    return default_dict_serializer(obj_dict,
+                                   key_transformer=key_transformer,
                                    strip_nulls=strip_nulls,
-                                   strip_privates=strip_privates, **kwargs)
+                                   strip_privates=strip_privates,
+                                   strip_properties=strip_properties,
+                                   strip_class_variables=strip_class_variables,
+                                   **kwargs)
 
 
-def _get_dict_from_obj(obj, strip_privates, cls=None, *_, **__):
+def _get_dict_from_obj(obj, strip_privates, strip_properties,
+                       strip_class_variables, cls=None, *_, **__):
     excluded_elems = dir(JsonSerializable)
+    props, other_cls_vars = _get_class_props(obj.__class__)
     return {attr: obj.__getattribute__(attr) for attr in dir(obj)
             if not attr.startswith('__')
             and not (strip_privates and attr.startswith('_'))
+            and not (strip_properties and attr in props)
+            and not (strip_class_variables and attr in other_cls_vars)
             and attr != 'json'
             and not isinstance(obj.__getattribute__(attr), Callable)
             and (not cls or attr in cls.__slots__)
             and attr not in excluded_elems}
 
 
+def _get_class_props(cls):
+    props = []
+    other_cls_vars = []
+    for n, v in _get_complete_class_dict(cls).items():
+        props.append(n) if type(v) is property else other_cls_vars.append(n)
+    return props, other_cls_vars
+
+
+def _get_complete_class_dict(cls):
+    cls_dict = {}
+    for parent_class in reversed(cls.mro()):
```

Comment:
The use of `reversed` here is a nice one! It is subtle though, and yet quite important. Maybe we should spend a comment that explains why it is there? Also, `parent_class` does not fully cover as `cls` itself is contained too.

How about something like this:

```suggestion
    # Loop reversed so values of sub-classes override those of super-classes.
    for cls_or_elder in reversed(cls.mro()):
```